app/controllers/api/v2/application_controller.rb

```diff
@@ -7,6 +7,7 @@ class ApplicationController < ::ApplicationController
 
       before_action :store_request_id
       before_action :check_terms_accepted
+      before_action :check_user_is_kept
```

Comment:
Should this run before the other actions here? Otherwise it sounds like we might tell them they need to accept terms and conditions when in reality they have no access at all.

Reply:
done